### ISSUE TITLE
Implement simple view methods for TensorAccessor/PackedTensorAccessor

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#60431]
THCDeviceTensor (the old version of PackedTensorAccessor) supports a bunch of handy utilities like transpose and view (and some more exotic ones like downcastInner). PackedTensorAccessor, the modern replacement, doesn't support any of these. They should!

**Comment by ramvenkat98** [comment]
Hi, I'm planning to implement these methods, and had some thoughts/questions about it.

Adding transpose and view to PackedTensorAccessor would be similar to what has been done for THCDeviceTensor (creating a new PackedTensorAccessor, with appropriately modified stride/size arrays); but, for TensorAccessor, we cannot do the same thing because we'd be mutating the underlying stride/size arrays of the original TensorAccessor as well which is undesirable.

A solution to this is to return a PackedTensorAccessor when methods like transpose() or view() are called from TensorAccessor - doing this, however, means that we should either only allow these methods to be called from host or allow PackedTensorAccessor to be constructed from devices as well. I'm not familiar with the reasoning behind why PackedTensorAccessor can currently only be constructed from the host - it makes sense to use it for transferring data from host to device, but is there a reason to not allow it on device?

Alternatively, for TensorAccessor, transpose() and view() could also be written as methods that only mutate the underlying sizes/strides and don't return a value - but I think this seems like it could easily lead to unsafe code, especially if there are multiple TensorAccessors lying around pointing to the same sizes/strides arrays.

I'm favoring returning a PackedTensorAccessor and allowing it to be constructed from the device - thoughts on this?

**Comment by ezyang** [comment]
This is absolutely right, you need an owning container, and only PackedTensorAccessor is that.

It is probably possible to make PackedTensorAccessor to be constructible on host, I'm not sure why it isn't setup that way but there may be some conceptual problem to make it happen.

As a bootcamp task, it's best to minimize the amount of changes you make in a single PR. So I suggest picking something that you think that will be easy, like just implementing transpose/view, and leave making PackedTensorAccessor constructible on device as a separate task.
